src/main/java/org/apache/commons/lang3/arch/Processor.java

```diff
@@ -0,0 +1,30 @@
+package org.apache.commons.lang3.arch;
+
+/**
+ *
+ */
+public class Processor {
+
+    private String name;
+    private ProcessorArch processorArch;
+    private ProcessorType processorType;
+
```

Comment:
I don't understand how the name field helps. 
It's obviously necessary to be able to get the Processor object from the name, but once you have the object, you know what the name is.
So why do you think the name is necessary?

The name field results in more objects being needed; if the name is not included then all the PPC+64 systems (e.g.) can share the same object in the init code.